addons/account/wizard/accrued_orders.py

```diff
@@ -105,28 +92,16 @@ def _get_computed_account(self, order, product, is_purchase):
         else:
             return accounts['income']
 
-    def _compute_move_vals(self):
-        def _get_aml_vals(order, balance, amount_currency, account_id, label="", analytic_distribution=None):
-            if not is_purchase:
-                balance *= -1
-                amount_currency *= -1
-            values = {
-                'name': label,
-                'debit': balance if balance > 0 else 0.0,
-                'credit': balance * -1 if balance < 0 else 0.0,
-                'account_id': account_id,
-            }
-            if analytic_distribution:
-                values.update({
-                    'analytic_distribution': analytic_distribution,
-                })
-            if len(order) == 1 and self.company_id.currency_id != order.currency_id:
-                values.update({
-                    'amount_currency': amount_currency,
-                    'currency_id': order.currency_id.id,
-                })
-            return values
+    def _get_aml_vals(self, label, balance, account_id, amount_currency=None, analytic_distribution=None, order=None, is_purchase=None, **kwargs):
+        if not is_purchase:
+            balance *= -1
+            amount_currency *= -1
+        if len(order) == 1 and self.company_id.currency_id != order.currency_id:
```

Comment:
Is `len(order) == 1` for testing that order is exactly one or that it is not `None` ?

Reply:
I didn't write this part of the code (I just moved it). So I'm not sure yet.
They are from [here](https://github.com/odoo/odoo/pull/134772/files/86a817fbeacbab1d3e69e6d1a09c275f77cca613#diff-35e373c2b9d7cb1321a3e45953b307949b9ab041c54c47c743e00e3cdee33538L123-L127)

Reply:
Let's find out then ! I think we can do something nice here :smile: